Source/WebKit/WebProcess/Extensions/API/Cocoa/WebExtensionAPINamespaceCocoa.mm

```diff
@@ -75,14 +75,12 @@
         return !extensionContext().supportsManifestVersion(3) && objectForKey<NSDictionary>(extensionContext().manifest(), @"page_action", false);
 
 #if ENABLE(WK_WEB_EXTENSIONS_SIDEBAR)
-    // FIXME: <https://webkit.org/b/276833> Check if the sidebar feature flag is enabled and if we have the sidePanel permission to determine if this property should be visible
+    // If the extension requests both sidePanel and sidebarAction, we will give them sidebarAction -- we check in sidePanel that there is no sidebar_action key, but we do not check in sidebarAction that there is no sidePanel permission
     if (name == "sidePanel"_s)
-        return false;
-
-    // FIXME: <https://webkit.org/b/276833> Check if the sidebar feature flag is enabled and if we have a sidebarAction key in the manifest to determine if this property should be visible
+        return page->corePage()->settings().webExtensionSidebarEnabled() && extensionContext().hasPermission("sidePanel"_s) && !objectForKey<NSDictionary>(extensionContext().manifest(), @"sidebar_action", true);
```

Comment:
Just for my notes - what is the `true` argument that's the last one passed to `objectForKey`? It looks like other callers of that method use `false` here.

Reply:
I think it is nil if empty when true, but I'm not looking at the header currently either. We should make these use an enum for this reason!

Reply:
Yep, the parameter is `nilIfEmpty`, which I think is more appropriate here since there is at least one sub-key to `sidebar_action` which is mandatory.